src/rust/plugin-registry/src/server/service.rs

```diff
@@ -109,16 +119,26 @@ pub struct PluginRegistry {
     config: PluginRegistryServiceConfig,
 }
 
-impl PluginRegistry {
-    fn ensure_artifact_size_limit(&self, bytes: &[u8]) -> Result<(), PluginRegistryServiceError> {
-        let limit_mb = &self.config.artifact_size_limit_mb;
-        if bytes.len() > (limit_mb * 1024 * 1024) {
-            Err(PluginRegistryServiceError::ArtifactTooLargeError(format!(
-                "Artifact exceeds {limit_mb}MB"
-            )))
-        } else {
-            Ok(())
+fn io_input_error(input: impl std::fmt::Display) -> std::io::Error {
+    std::io::Error::new(std::io::ErrorKind::InvalidInput, input.to_string())
+}
+
+/// Keep adding the size of an incoming chunk to an AtomicUsize.
+/// Start throwing errors when it exceeds limit.
+async fn accumulate_stream_size_to_limit(
+    stream_length: Arc<AtomicUsize>,
```

Comment:
I suspect you can just take `stream_length: &AtomicUsize`, and then call `.as_ref()` on the Arc. NBD but saves an atomic increment. I haven't seen the calling clone yet so I could be wrong.

Reply:
Based on the caller it might end up being a pain actually, so nvm, not worth it.